### ISSUE TITLE
Report : page numbering

### ISSUE THREAD

**Issue by frgis** [qgis/QGIS#33636]
Hi,

Currently, it does not seem possible to add page numbering at the bottom of a page to a report.
Would it be possible to add this possibility?

Thank you,

**Comment by haubourg** [comment]
Agreed, much needed. If anyone can support this, it will be more than welcom

**Comment by frgis** [comment]
If I had developer skills I would participate well. Unfortunately I don't have one. But I started to translate the documentation into French as a little while during my free time :-)

**Comment by nicolasmcfadden** [comment]
I echo frgis, i sadly don't have the coding skills... is there a plan to implement this? Thanks for all the work. I'm a big fan of qgis

**Comment by nicolasmcfadden** [comment]
Hello, any updates on this?

**Comment by djes** [comment]
Hi, atlas is concerned too. I just discovered that it's not possible to create dynamic names for the pages of an atlas. The name is evaluated once and a page number is just added at the end of the name. 

These parameters
![image](https://github.com/qgis/QGIS/assets/12065874/a7633c46-a7f8-4d99-8695-996cd341b107)

gives

![image](https://github.com/qgis/QGIS/assets/12065874/b0cb91a6-f804-4f9d-9d5b-0c1f96e9070f)

Could it be added in future versions ? Thank you !

**Comment by jermfCH** [comment]
Hi,

Still no solution to this?
It's a bit sad — this functionality seems so basic and would be very useful.

I have to say, the Report feature in QGIS is really useful, but it still feels a bit unfinished. I’ve encountered quite a few bugs or had to rely on workarounds to produce good-quality reports.

I really hope this gets improved in the future!

**Comment by haubourg** [comment]
> It's a bit sad — this functionality seems so basic and would be very useful.

As usual in open source software, things do not happen by magic. QGIS is user and community driven. 
So if this feature  is on your  critical path, I suggest you try to gather funds to make this happen. There are several ways to do it, from participating to you national user group and sponsor features, to convince you company to fund this feature and sustain QGIS.org, to the most difficult path, start coding yourself ( which I suggest only if you are a passionate developper ready to jump in to C++ and high quality standards software with review process) .

**Comment by HP2-be** [comment]
The trick is to add the numbers once the document is final - so I guess it might be more a PDF setting than a QGIS thing?

I have no programming skills either, but wonder whether it's possible to enhance the PDF creator options to include page numbering in the end after the Report is created. Adding options for keeping URL links active goes with it as well. Last is probably already covered in another ticket, I realize just now.
